backend/main.py

```
@@ -38,30 +74,183 @@
+        # Resize large images for better performance
+        img = Image.open(file.file)
+        if img.width > 1024 or img.height > 1024:
+            # Calculate new size maintaining aspect ratio
+            ratio = min(1024 / img.width, 1024 / img.height)
+            new_width = int(img.width * ratio)
+            new_height = int(img.height * ratio)
+
+            img = img.resize((new_width, new_height), Image.Resampling.LANCZOS)
+
+            # Save resized image back to file
+            output = io.BytesIO()
```

Ruff (0.14.14)

[error] Undefined name `io`

(F821)